cfme/tests/infrastructure/test_timelines.py

```diff
@@ -30,213 +25,161 @@ def a_provider(request):
-
-    logger.info("found events: {evt}".format(evt="\n".join([repr(e) for e in found_events])))
-    return len(found_events)
-
-
-def test_infra_provider_event(gen_events, new_vm):
-    """Tests provider event on timelines
-
-    Metadata:
-        test_flag: timelines, provision
-    """
-
-    wait_for(count_events, [new_vm.provider, new_vm], timeout='5m', fail_condition=0,
-             message="events to appear")
+    vm = VM.factory(random_vm_name('timelines', max_length=16), a_provider)
+    logger.info('Fixture new_vm set up! Name: %r Provider: %r', vm.name, vm.provider.name)
+    yield vm
+    logger.info('Fixture new_vm teardown! Name: %r Provider: %r', vm.name, vm.provider.name)
 
 
-def test_infra_host_event(appliance, a_provider, gen_events, new_vm):
-    """Tests host event on timelines
+class VMEvent(object):
+    """Class for generating  events on a VM in order to check it on Timelines. According to the
+    event passed in the parameter it will perform the action on the given VM.
 
-    Metadata:
-        test_flag: timelines, provision
+    Args:
+        vm: A VM object (Object)
+        event: An event which will be generated (string).
     """
-    view = navigate_to(new_vm, "Details")
-    host_name = view.entities.relationships.get_text_of('Host')
-    host_collection = appliance.collections.hosts
-    host = host_collection.instantiate(name=host_name, provider=a_provider)
-    wait_for(count_events, [host, new_vm], timeout='10m', fail_condition=0,
-             message="events to appear")
 
+    def __init__(self, vm, event):
+        self.vm = vm
+        self.action = event
+
+        if self.action == 'create':
+            self.tl_event = ('VmDeployedEvent', 'USER_RUN_VM')
+            self.tl_category = 'Creation/Addition'
+            self.db_event_type = ('vm_create', 'USER_RUN_VM')
+            self.emit_cmd = self._setup_vm
+        elif self.action in {'start', 'resume'}:
+            self.tl_event = ('VmPoweredOnEvent', 'USER_STARTED_VM')
+            self.tl_category = 'Power Activity'
+            self.db_event_type = 'vm_start'
+            self.emit_cmd = self._power_on
+        elif self.action == 'stop':
+            self.tl_event = ('VmPoweredOffEvent', 'USER_STOP_VM')
+            self.tl_category = 'Power Activity'
+            self.db_event_type = 'vm_poweroff'
+            self.emit_cmd = self._power_off
+        elif self.action == 'suspend':
+            self.tl_event = ('VmSuspendedEvent', 'USER_SUSPEND_VM')
+            self.tl_category = 'Power Activity'
+            self.db_event_type = 'vm_suspend'
+            self.emit_cmd = self._suspend
+        elif self.action == 'rename':
+            self.tl_event = 'VmReconfiguredEvent'
+            self.tl_category = 'Configuration/Reconfiguration'
+            self.db_event_type = 'VmRenamedEvent'
+            self.emit_cmd = self._rename_vm
+        else:
+            assert False, logger.warn('%r is not a proper parameter.', self.action)
```

Comment:
I don't think this else statement is necessary but If you think it is needed, please replace it with something like
`
raise ValueError() 
`